blogposts/2020/evolution-of-the-precise-code-intel-backend.md

```diff
@@ -0,0 +1,175 @@
+---
+title: "Evolution of the precise code intel backend"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-06-17T10:00-07:00
+tags: [blog]
+slug: evolution-of-the-precise-code-intel-backend
+published: true
+---
+
+On July 12, Sourcegraph's LSIF-based precise code intelligence will have received its first commit a year ago.
+
+**🎉 Happy 🎉 Almost 🎉 Birthday 🎉**
+
+This post reflects on the high-level technical changes as the services providing the feature matured through additional features, changing environment requirements, hardening, performance improvements, refactoring, and one major rewrite in a different language. These changes span +324k/-277k line of code over 527 commits.
+
+I may not _strictly_ be able to call this post a work of [software archaeology](https://en.wikipedia.org/wiki/Software_archaeology) (since it's only a year old, it's actively used, not a legacy codebase, the primary author is still here to talk about it, and it isn't completely undocumented). It may be more accurate to call it an [historiographical artifact](https://en.wikipedia.org/wiki/Historiography), but dinosaurs are cooler than books so welcome to the dig site.
+
+## Provenance
+
+**PR**: [Add LSIF support (#4799)](https://github.com/sourcegraph/sourcegraph/pull/4799)
+
+Christ Wendt wrote the initial draft of the service as a simple TypeScript express server proxied by the frontend service's HTTP API. This kept the number of public services low and kept request authenticate flow to a single code path. The express server would accept raw LSIF input and store it on disk, unchanged. Queries to the lsif-server via a proof of concept LSIF-specific [extension](https://docs.sourcegraph.com/extensions) would read the raw LSIF data for that repository, parse it into memory, and walk the graph to construct the appropriate response.
+
+![architecture diagram](https://storage.googleapis.com/sourcegraph-assets/lsif-arch-1.png)
+
+The choice of TypeScript was a natural one at the time: the (then called) codenav team consists of what would later become the web and the code intelligence teams. TypeScript was a major core competency of the team, and there was an additional benefit of being able to reuse code from the [server](https://github.com/microsoft/vscode-lsif-extension/tree/d9d4f25ffcd735d6a45bdfe0f3c811cf4bb376e0/server/src) package in an unpublished Visual Studio Code [extension](https://github.com/microsoft/vscode-lsif-extension) to head-start the service with a functional consumer of the LSIF protocol (which was only 0.4.0 at this point in time).
```

Comment:
head-start -> jumpstart?

Reply:
Ah yeah I did several edits and didn't realize I was using it as a verb. Thanks!